Add SSR html body ad injection components.

B77Mills opened the pull request:
This ads the components needed to be able to pass in overrides for the html ad injections into the body and transcript fields.

**Example PR using this new component:** https://github.com/parameter1/randall-reilly-websites/pull/498

**Example PR using a spin off of this new component:** https://github.com/parameter1/ab-media-websites/pull/248

Thing to note is the dataPropToPreventDups prop.  This is what the ads currently use to see if the sibling html element is the same or in the case of ads if the zone or GAMTemplate match.  Prevent them from putting two identical ad calls side by side. 
```js
$ const gamAdInjection = (GAM) ? [
  {
    counts: desktopAdCounts,
    name: desktopAdName,
    modifiers: desktopAdModifiers,
    dataPropToPrevetDups: desktopAdDataPropToPrevetDups
  },
  {
    counts: mobileLeaderboardAdCounts,
    name: mobileLeaderboardAdName,
    modifiers: mobileLeaderboardAdModifiers,
    dataPropToPrevetDups: mobileLeaderboardAdDataPropToPrevetDups
  },
  {
    counts: mobileAdCounts,
    name: mobileAdName,
    modifiers: mobileAdModifiers,
    dataPropToPrevetDups: mobileAdDataPropToPrevetDups
  }
] : [];

```

Produces the following config:
```json
[
  {
    "counts": [
      1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500],
    "name": 'inline-content-desktop',
    "modifiers": ['margin-auto-x', 'inline-content', 'inline-content-desktop'],
    "dataPropToPrevetDups": 'gamTemplateName'
  },
  {
    "counts": [250],
    "name": 'inline-leaderboard-mobile',
    "modifiers": ['margin-auto-x', 'inline-content', 'inline-leaderboard-mobile'],
    "dataPropToPrevetDups": 'gamTemplateName'
  },
  {
    "counts": [900, 1650, 2950, 4250, 5550, 6850, 8150, 9450, 10750, 12050, 13350, 14650, 15950, 17250],
    "name": 'inline-content-mobile',
    "modifiers": ['margin-auto-x', 'inline-content', 'inline-content-mobile'],
    "dataPropToPrevetDups": 'gamTemplateName'
  }
]
```

B77Mills committed:
Add SSR html body ad injection components.

This ads a the components needed to be able to pass in a adInjectionConfig to be able to configure which ad showes up at which conut with a list of modifiers.

The default ad configs is:
```js
$ const adInjectionConfig = defaultValue(input.adInjectionConfig, [
  {
    name: 'inline-content-desktop',
    modifiers: ["margin-auto-x",  "inline-content", "inline-content-desktop"],
    counts: [1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500],
  },
  {
    name: 'inline-content-mobile',
    modifiers: ["margin-auto-x",  "inline-content", "inline-content-mobile"],
    counts: [250, 900, 1650, 2950, 4250, 5550, 6850, 8150, 9450, 10750, 12050, 13350, 14650, 15950, 17250],
  }
]);
```

B77Mills committed:
Convert bodyWithInjection template to handle new injection prop

Swith adInjectionConf to be injections with is a array of objects with count and html in it.  This will allow you to pass in the html or the component you would like to render & the counts you would like it to render.  New default is as follows:
```
$ const injections = defaultValue(input.adInjectionConfig, [
  {
    counts: [1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500],
    html: GAMDefineDisplayAd.renderToString({
      name: 'inline-content-desktop',
      modifiers: ["margin-auto-x",  "inline-content", "inline-content-desktop"],
      $global
    })
  },
  {
    counts: [250, 900, 1650, 2950, 4250, 5550, 6850, 8150, 9450, 10750, 12050, 13350, 14650, 15950, 17250],
    html: GAMDefineDisplayAd.renderToString({
      name: 'inline-content-mobile',
      modifiers: ["margin-auto-x",  "inline-content", "inline-content-mobile"],
      $global
    })
  }
]);
```

B77Mills committed:
update prop to get defaultValue off of

zarathustra323 commented on packages/marko-web-theme-monorail/components/ssr-html-inject.marko:
```diff
@@ -0,0 +1,96 @@
+            const r = $.parseHTML(contents);
+            const [toInsert] = $.parseHTML(contents);
+            if (!toInsert) return;
+            if ($(toInsert).data() && $(toInsert).data().bamZoneAlias && $(toInsert).data().bamZoneAlias === $next.data().bamZoneAlias) return;
```
this is only account for Broadstreet, not GAM... will need to determine how to handle both.

B77Mills committed:
Add GAM check to only add ad injection if GAM is set

B77Mills committed:
Remove default injection ref to gam

No longer assume the group is using gam for ads and rather assume that the injections is an empty array or is being passed in.

B77Mills committed:
Add the ability to prevent dups by dataKey

if you pass in a dataPropToPreventDups prop it will try to prevent the injection of two identical items.

B77Mills committed:
test

B77Mills committed:
test

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [CLS: Body copy shifting when ads load on mobile](https://trello.com/c/aQnuynEI/193-cls-body-copy-shifting-when-ads-load-on-mobile)

B77Mills committed:
Set body with injection to use default GAM ad injection

update the body-with-injection.marko component to use default values and components for GAM ad injections.  Also enabled on example website.

Now by default it will have the following:
 - Desktop Inline Ad @ [1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500]
 - Mobile Leaderboard(Standard mobile inline ad name & modifiers by default) @ [250]
 - Mobile Inline Ad @ [900, 1650, 2950, 4250, 5550, 6850, 8150, 9450, 10750, 12050, 13350, 14650, 15950, 17250]

brandonbk commented on packages/marko-web-theme-monorail/components/content/body-with-injection.marko:
```diff
@@ -0,0 +1,77 @@
+$ const desktopAdCounts = defaultValue(input.desktopAdCounts, [1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500]);
+$ const desktopAdName = defaultValue(input.desktopAdName, "inline-content-desktop");
+$ const desktopAdModifiers = defaultValue(input.desktopAdModifiers, ["margin-auto-x",  "inline-content", "inline-content-desktop"]);
+$ const desktopAdDataPropToPrevetDups = defaultValue(input.desktopAdDataPropToPrevetDups, "gamTemplateName");
```
@B77Mills - replace `PrevetDups` with `PreventDupes` in all files.  Missing the letter "n" in Prevent, and "e" in Dupes.

B77Mills committed:
Fix prop name

Fix spelling of prop name
